annnufan/assignment3/test_words_finder.cpp

```diff
@@ -0,0 +1,39 @@
+#include <iostream>
+#include "codeu_test_lib.h"
+#include "words_finder.cpp"
+
+std::vector<std::vector<char>> ex_grid({{'a', 'a', 'r'}, {'t', 'c', 'd'}});
+std::vector<std::vector<char>> empty_grid;
+std::vector<std::vector<char>> grid_1({{'a', 'o', 'b', 'e', 'r'}, {'c', 'a', 'd', 'e', 't'}, {'b', 'i', 'v', 'i', 'n'}});
+
+dictionary ex_dict({"car", "card", "cart", "cat"});
+dictionary empty_dict;
+dictionary dict_1({"tree", "cart", "bee", "bed", "cider", "beer", "teer", "dinner", "baobab", "vine", "tear", "horse", "bob"});

```

Comment:
If not shared between tests, it is recommended to define test input and expected output inside the test case. This way the reader can see the entire test in its context without the need to scroll up and down:
```C++
void test_double_usage_cell() {
    std::vector<std::vector<char>> grid = {{'s', 'o', 'm', 'e'}
                                           {'g', 'r', 'i', 'd'}};
    dictionary dict({"some", "words"});
    std::set<std::string> expected_result = {...};
    EXPECT_EQ(expected_result, find_words_from_dict_on_grid(dict, grid));
}
```

Even if the input IS shared between tests, some say it is ok to copy-paste it to all the tests so each test will be easily readable by itself. (Yes, code duplication is more forgivable in tests!)